app/models/programme.rb

```diff
@@ -32,6 +32,15 @@ class Programme < ApplicationRecord
     event :complete do
       after do
         self.update(ended_on: Time.zone.now)
+        if EnergySparks::FeatureFlags.active?(:activities_2023)
```

Comment:
This doesn't meet the specification: "When a school completes a programme, within the same academic year as they started it, then they should be awarded the bonus points."

This code will award them points whenever it was started. You'll need to use the schools calender to find the academic year for the programme start date and then check whether its the same for the completion date. See: `school.academic_year_for(...)`

Reply:
@ldodds pushed a couple of updates - should (I think, anyway) now check and only create an `Observation` when the school completes a programme within the same academic year as the start date (couple of tests for the same added too).

Reply:
@iant sorry, I maybe I should have been clearer:

- its useful to mark the completion of the programme in the school timeline, regardless of the points. So the Observation can be created when the Programme is complete
- they should only get the bonus points, if its completed in the same academic year

So I think you just need to use the check for current academic year to decide whether to award zero points or the bonus points.

Reply:
Ah, ok.  sorry for my misunderstanding :)  no probs at all.  have pushed an update so an observation is still created but points only awarded if the programme is completed in same academic year